src/main.rs

```diff
@@ -45,12 +45,8 @@ fn main() {
         .expect("unable to set SIGINT handler");
 
         log::debug!("beginning normal operation...");
-        let service = server::serve(&args);
+        server::serve(&args)?;
         exiting.join().expect("unable to join SIGINT handler thread");
-        if service.is_err() {
-            log::error!("unable to run server: {}", service.unwrap_err());
-            std::process::exit(4);
```

Comment:
This exit-code is important to help determine whether an environmental error has occurred. I must ask that it not be suppressed -- the changes here would replace it with `1`.